packages/legend-application/src/components/shared/BasicValueSpecificationEditor.tsx

```diff
@@ -170,23 +171,83 @@ const StringPrimitiveInstanceValueEditor = observer(
     className?: string | undefined;
     setValueSpecification: (val: ValueSpecification) => void;
     resetValue: () => void;
+    valueOptions?:
+      | {
+          options: (string | number)[] | undefined;
+          isLoading: boolean;
+          updateOptions: () => void;
+        }
+      | undefined;
   }) => {
-    const { valueSpecification, className, resetValue, setValueSpecification } =
-      props;
+    const {
+      valueSpecification,
+      className,
+      resetValue,
+      setValueSpecification,
+      valueOptions,
+    } = props;
+    const useSelector = Boolean(valueOptions);
     const value = valueSpecification.values[0] as string;
-    const changeValue: React.ChangeEventHandler<HTMLInputElement> = (event) => {
-      instanceValue_changeValue(valueSpecification, event.target.value, 0);
+    const updateValueSpec = (val: string): void => {
+      instanceValue_changeValue(valueSpecification, val, 0);
       setValueSpecification(valueSpecification);
     };
+    const changeInputValue: React.ChangeEventHandler<HTMLInputElement> = (
+      event,
+    ) => {
+      updateValueSpec(event.target.value);
+    };
+    // custom select
+    const selectedValue = { value: value, label: value };
+    const changeValue = (
+      val: null | { value: number | string; label: string },
+    ): void => {
+      const newValue = val === null ? '' : val.value.toString();
+      updateValueSpec(newValue);
+    };
+    const handleInputChange = (
+      inputValue: string,
+      actionChange: InputActionMeta,
+    ): void => {
+      if (actionChange.action === 'input-change') {
+        updateValueSpec(inputValue);
+        valueOptions?.updateOptions();
+      }
+    };
+    const isLoading = valueOptions?.isLoading;
+    const queryOptions = valueOptions?.options?.length
+      ? valueOptions.options.map((e) => ({
+          value: e,
+          label: e.toString(),
+        }))
+      : undefined;
+    const noOptionsMessage =
+      valueOptions?.options === undefined ? (): null => null : undefined;
+
     return (
       <div className={clsx('value-spec-editor', className)}>
-        <input
-          className="panel__content__form__section__input value-spec-editor__input"
-          spellCheck={false}
-          value={value}
-          placeholder={value === '' ? '(empty)' : undefined}
-          onChange={changeValue}
-        />
+        {useSelector ? (
+          <CustomSelectorInput
+            className="value-spec-editor__enum-selector"
+            options={queryOptions}
+            onChange={changeValue}
+            value={selectedValue}
+            onInputChange={handleInputChange}
+            darkMode={true}
+            isLoading={isLoading}
+            allowCreateWhileLoading={true}
+            autoFocus={true}
+            noOptionsMessage={noOptionsMessage}
+          />
+        ) : (
+          <input
```

Comment:
In terms of styling, from the video, I notice the discrepancy in height of the custom selector component and the height of its container causes a jump when users input, let's fix that.

Also, I think we should consider whether we should show the `caret down` here as it suggests auto-complete is available as people type and it's sort of distracting. What we want to do here is just let people have the feeling like they are typing some text value per usual, and happen to have these suggestions popping up.